src/m3ninx/index/segment/fst/fst_terms_postings_iterator.go

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package fst
+
+import (
+	sgmt "github.com/m3db/m3/src/m3ninx/index/segment"
+	"github.com/m3db/m3/src/m3ninx/postings"
+	postingsroaring "github.com/m3db/m3/src/m3ninx/postings/roaring"
+	"github.com/pilosa/pilosa/roaring"
+)
+
+// postingsIterRoaringPoolingConfig uses a configuration that avoids allocating
+// any containers in the roaring bitmap, since these roaring bitmaps are backed
+// by mmaps and don't have any native containers themselves.
+var postingsIterRoaringPoolingConfig = roaring.ContainerPoolingConfiguration{
```

Comment:
I think if you don't want any containers you can just use the non pooling one?

Reply:
We want the containers to be reused when Reset() is called on them and then they are used again though.  

Otherwise it starts allocating no?   I thought if we don't specify `NewBitmapWithPooling(...)` that containers never get returned when a call to reset is used?  We reuse a single bitmap very frequently since we're iterating over all terms from an existing FST segment (call UnmarshalBinary on each terms over and over again) as we're iterating over the entire FST and reading the postings list for every term in the segment.

Reply:
Ok cool you want the pooling. What I'm saying is with the configuration you have:

```golang
var postingsIterRoaringPoolingConfig = roaring.ContainerPoolingConfiguration{
 	AllocateArray:                   false,
	AllocateRuns:                    false,
	AllocateBitmap:                  false,
	MaxCapacity:                     128,
	MaxKeysAndContainersSliceLength: 128 * 10,
}
```

Every call to Reset will discard all the containers. If you want to keep them around (and there is only one bitmap so you don't mind it getting kind of big) I would do something like:

```golang
var postingsIterRoaringPoolingConfig = roaring.ContainerPoolingConfiguration{
 	AllocateArray:                   true,
	AllocateRuns:                    true,
	AllocateBitmap:                  true,
	MaxCapacity:                     128,
	MaxKeysAndContainersSliceLength: 128 * 10,
}
```

Although I'd like to get this P.R merged in: https://github.com/m3db/pilosa/pull/4/

and then you can just do:

```golang
roaring.NewContainerWithDefaultPoolingConfiguration(128)
```